### DISCUSSION TITLE
When useQuery executeQuery always the same result

### DISCUSSION THREAD

**Discussion by Politanskyi** [urql-graphql/urql#2096] [Q&A] [answered]
After a successful request, I want to get more results using the function executeQuery, but even when changing the parameters, I always get the same result, it feels like it just won't accept the passed parameters!

```
const salesLimit = 10;
const page = ref(1);
const moreLoadingKey = ref(false);
const hasMorePages = ref(true);
const queryVariables = { limit: salesLimit, page: page.value };

const { data, fetching, error, executeQuery } = await useQuery({
	query: GET_SALES_PAGE,
	variables: queryVariables,
	requestPolicy: 'cache-and-network'
});

sales.value = data.value.sales.data;

const showMore = () => {
	page.value += 1;

	console.log(page.value)
	executeQuery({
		variables: { page: page.value },
		requestPolicy: 'network-only'
	})
};

```

![image](https://user-images.githubusercontent.com/43199789/143271062-13964eb0-c9da-497d-8aeb-ab8e7f6ba1c3.png)

**Comment by JoviDeCroock** [comment, marked as answer]
We do not support passing in `variables` in the `executeQuery` currently 😅 [it accepts OperationContext](https://formidable.com/open-source/urql/docs/api/urql/#usequery)

**Comment by Politanskyi** [comment]
> We do not support passing in `variables` in the `executeQuery` currently 😅 [it accepts OperationContext](https://formidable.com/open-source/urql/docs/api/urql/#usequery)

How then to send a request with new parameters?

**Comment by JoviDeCroock** [comment]
update the `variables` in your `useQuery` and it will update. There are also some common UI-patterns that you can use [here](https://formidable.com/open-source/urql/docs/basics/ui-patterns/#infinite-scrolling)

Generally however in React we do:

```js
const Component = () => {
  const [page, setPage] = useState(1)
  const [data] = useQuery({ query: x, variables: { page } })
  
  const nextPage = () => setPage(page + 1)
}
```

Or if you consider the case for Vue just increment the `ref` which causes a rerender and the `useQuery` will reexecute. (didn't notice you were using vue)

Here's a vue-example: https://github.com/FormidableLabs/urql/blob/main/examples/with-vue3/src/PokemonList.vue

**Comment by Politanskyi** [comment]
But how marge result??

**Comment by JoviDeCroock** [comment]
You can leverage a normalized-caching resolver, an intermediary state with an effect or a similar pattern as the one linked on ui-patterns.

**Comment by Politanskyi** [comment]
> You can leverage a normalized-caching resolver, an intermediary state with an effect or a similar pattern as the one linked on ui-patterns.

When i do this page is change to second, but result go to the last part 
![image](https://user-images.githubusercontent.com/43199789/143276322-7f0ff6b2-3c10-4cbe-bae5-3d75c476786d.png)
![image](https://user-images.githubusercontent.com/43199789/143276360-2cfbc0ae-0a06-4fa9-91c9-e8f094ee057a.png)
but i have 100 posts
![image](https://user-images.githubusercontent.com/43199789/143276463-e6da1dc6-45ed-496b-b796-a95c678b6454.png)

**Comment by kitten** [comment]
There's several patterns you can apply for infinite scrolling:
- Split it up into page chunks, i.e. one query per "page" and merge the components into a parent
- Let Graphcache, i.e. our normalised cache, do a quick combination of all pages into one big list
- Manually update the state
We have docs and examples for all of these
- [One component per page: docs](https://formidable.com/open-source/urql/docs/basics/ui-patterns/#infinite-scrolling)
- [One component per page: example](https://github.com/FormidableLabs/urql/tree/main/examples/with-pagination)
- [Quick pagination with Graphcache: docs](https://formidable.com/open-source/urql/docs/graphcache/local-resolvers/#pagination)
- [Quick pagination with Graphcache: example](https://github.com/FormidableLabs/urql/tree/main/examples/with-graphcache-pagination)

Graphcache requires your schema to look a certain way for pagination, if you don't want to write your own resolver. It's also harder to invalidate results this way.

The component per page approach is much quicker but you'll have to rewrite it for Vue. However, it's not only the simplest approach, but often the easiest if you don't need the data in one place (which you'd really only need on React Native afaik)
